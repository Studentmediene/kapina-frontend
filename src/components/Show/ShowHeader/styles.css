.container {
  display: flex;
  flex-direction: column;
  padding-bottom: 2rem;
  width: 100%;
}

.imageWrapper {
  width: 100%;
}

.image {
  width: 100%;
}

.showText {
  width: 100%;
}

.categories {
  padding-top: 1rem;
  color: gray;
  width: 100%;
}

.title {
  text-align: center;
}

@media (min-width: 1000px) {
  .container {
    flex-direction: row;
    flex-wrap: wrap;
  }

  .imageWrapper {
    padding-right: 1rem;
    width: 35%;
  }

  .showText {
    display: flex;
    flex-direction: column;
    padding-left: 1rem;
    width: 65%;
  }

  .title {
    text-align: left;
  }
}
