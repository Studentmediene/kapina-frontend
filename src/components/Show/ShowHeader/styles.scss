@import 'variables.scss';

.container {
  display: flex;
  flex-direction: column;
  padding-bottom: 32px;
  width: 100%;
}

.imageWrapper {
  width: 100%;
}

.image {
  width: 100%;
}

.showText {
  width: 100%;
}

.categories {
  color: gray;
}

.podcastLink,
.categories {
  padding-top: 1rem;
  width: 100%;
}

.title {
  text-align: center;
}

@media screen and (min-width: $breakpoint-small) {
  .container {
    flex-direction: row;
    flex-wrap: wrap;
    align-items: center;
  }

  .imageWrapper {
    padding-right: 16px;
    width: 35%;
  }

  .showText {
    display: flex;
    flex-direction: column;
    padding-left: 1rem;
    width: 65%;
  }

  .title {
    text-align: left;
  }
}
