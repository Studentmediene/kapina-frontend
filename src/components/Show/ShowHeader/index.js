import React from 'react';
import PropTypes from 'prop-types';

import styles from './styles.scss';
import 'components/common/styles/editor.scss';

const ShowHeader = ({
  show: {
    show: { title, content, logoImageUrl, categories },
  },
}) => {
  let categoryTags;
  if (categories && categories.length > 0) {
    categoryTags = (
      <div className={styles.categories}>
        <span>Kategorier: </span>
        {categories.map((category, index) => (
          <span key={index}>{category.name}</span>
        ))}
      </div>
    );
  }
  return (
    <div className={styles.container}>
      <div className={styles.imageWrapper}>
        <img className={styles.image} src={logoImageUrl} alt={title} />
      </div>
      <div className={styles.showText}>
        <h2 className={styles.title}>{title}</h2>
        <div dangerouslySetInnerHTML={{ __html: content }} />
      </div>
      {categoryTags}
    </div>
  );
};

ShowHeader.propTypes = {
  show: PropTypes.oneOfType([PropTypes.bool, PropTypes.object]),
};

export default ShowHeader;
