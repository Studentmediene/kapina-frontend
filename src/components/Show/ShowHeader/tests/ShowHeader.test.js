import React from 'react';
import { shallow } from 'enzyme';

import ShowHeader from '../';

describe('<ShowHeader />', () => {
  it('renders correctly', () => {
    const mockShow = {
      title: 'Title',
      content: '<p>Content</p>',
      logoImageUrl: 'Image url',
      categories: [{ name: 'category1' }, { name: 'category2' }],
    };

    const tree = shallow(<ShowHeader show={mockShow} />);
    expect(tree).toMatchSnapshot();
  });
});
