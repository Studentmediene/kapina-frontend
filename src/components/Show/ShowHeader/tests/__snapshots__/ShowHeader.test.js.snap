// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowHeader /> renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="imageWrapper"
  >
    <img
      alt="Title"
      className="image"
      src="Image url"
    />
  </div>
  <div
    className="showText"
  >
    <h2
      className="title"
    >
      Title
    </h2>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Content</p>",
        }
      }
    />
  </div>
  <div
    className="categories"
  >
    <span>
      Kategorier: 
    </span>
    <span
      key="0"
    >
      category1
    </span>
    <span
      key="1"
    >
      category2
    </span>
  </div>
</div>
`;
